File: TPAWebApi/Libs/BusinessLogic/Models/Appointment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace BusinessLogic.Models
{
    public class Appointment : BaseModel
    {
        [Required]
        public string EntryName { get; set; }
        public string EntryText { get; set; }
        public DateTime DateDue { get; set; }
        [Required]
        public DateTime DateCreated { get; set; }
        public short Duration { get; set; }
        [Required]
        public bool Status { get; set; }
        [Required]
        public int UserId { get; set; }
    }
}

File: TPAFrontEnd/src/models/appointment.ts
export class Appointment {
   id: number;
   entryName: string;
   entryText: string;
   dateDue: Date;
   dateCreated: Date;
   duration: number;
   status: boolean;
   userID: number;
   constructor(values: Object = {}) {
        Object.assign(this, values);
   }
}